python/caffe/layers/detectnet/mean_ap.py

```diff
@@ -28,22 +26,31 @@ class 1 - true positive, 2 - false positive, 3 - true negative
         python_param {
             module: 'caffe.layers.detectnet.mean_ap'
             layer: 'ScoreDetections'
+            # parameter - IoU threshold (optional - default: 0.7)
+            param_str: '0.7'
         }
         include: { phase: TEST }
     }
     """
+    
 
     def setup(self, bottom, top):
-        pass
+        # Default IoU threshold
+        self.iou_threshold = 0.7 
+        try:
+            self.iou_threshold = float(self.param_str)
```

Comment:
I wonder if we should head off the annoyance of adding params to the end of the list and just make this a JSON parameter instead of a list (e.g. https://gist.github.com/lukeyeager/5ae183044d96ef1540f2).

Pros:
* Each parameter is clearly named in prototxt
* Parameter is more flexible to future changes and especially to merging multiple changes in from different branches

Cons:
* Doesn't match the format of params for other layers

What do you think?